pkg/restore/restore.go

```diff
@@ -658,16 +658,37 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		_, restoreErr := resourceClient.Create(obj)
 		if apierrors.IsAlreadyExists(restoreErr) {
 			equal := false
-			if fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{}); err == nil {
+			fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+			if err == nil {
 				equal, err = objectsAreEqual(fromCluster, obj)
 				// Log any errors trying to check equality
 				if err != nil {
-					ctx.infof("error checking %s against cluster: %v", obj.GetName(), err)
+					ctx.infof("error checking %s against cluster: %v", kube.NamespaceAndName(obj), err)
 				}
 			} else {
-				ctx.infof("Error retrieving cluster version of %s: %v", obj.GetName(), err)
+				ctx.infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
 			}
 			if !equal {
+				if groupResource == kuberesource.ServiceAccounts {
+					desired, err := mergeServiceAccounts(fromCluster, obj)
+					if err != nil {
+						ctx.infof("error merging secrets for ServiceAccount %s: %v", kube.NamespaceAndName(obj), err)
+					}
+					patchBytes, err := generatePatch(fromCluster, desired)
+					if err != nil {
+						ctx.infof("error generating patch for ServiceAccount %s: %v", kube.NamespaceAndName(obj), err)
```

Comment:
we should probably add a warning or error here (like https://github.com/heptio/ark/pull/403/files#diff-7e2ddda55b692c82f9fa5f89a3d05e54R687)